File: Programmers/Level3/단속카메라.py
def solution(routes):
    answer = 0
    routes.sort(key=lambda x: -x[1])

    while routes:
        # 단속 카메라 설치
        target = routes.pop()[1]
        answer += 1
        # 단속 대상 확인
        i = 0
        while routes and i < len(routes):
            if routes and routes[i][0] <= target <= routes[i][1]:
                routes.pop(i)
            else:
                i += 1

    return answer


# 다른 풀이 ----------
def solution2(routes):
    answer = 0
    routes.sort(key=lambda x: x[1])

    point = -30001
    for rs, re in routes:
        if point < rs:
            answer += 1
            point = re

    return answer


routes = [[-20,15], [-14,-5], [-18,-13], [-5,-3]]
print(solution(routes))

File: 프로그래머스/3/42884. 단속카메라/단속카메라.java
import java.util.Arrays;

class Solution {
    public int solution(int[][] routes) {
        int answer = 0;
        int target = Integer.MIN_VALUE;
        
        Arrays.sort(routes, (a, b) -> a[1] - b[1]);
        
        for (int[] route : routes) {
            if (target < route[0]) {
                answer++;
                target = route[1];
            }
        }
        
        return answer;
    }
}